### ISSUE TITLE
Error with optional chaining in 3.1.0: "TypeError: Cannot read properties of null (reading 'getLiteralValueAtPath')"

### ISSUE THREAD

**Issue by ecraig12345** [rollup/rollup#4810]
### Rollup Version

3.10.0 (worked in 3.9.1)

### Operating System (or Browser)

Mac OS, browser

### Link To Reproduction

https://rollupjs.org/repl/?version=3.10.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHV0aWwlMjBmcm9tJTIwJy4lMkZ1dGlsJyUzQiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMHV0aWwuZm9vJTNGLmJhciUzQiU1Q25jb25zb2xlLmxvZyhiYXIpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydXRpbC5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMCU3QiUyMGJhciUzQSUyMCU3QiU3RCUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

### Expected Behaviour

The following code which uses optional chaining should bundle without errors. (This is a contrived example; the real code involves reading from objects where a key won't always exist.)

index.js
```js
import * as util from './util';
const bar = util.foo?.bar;
console.log(bar);
```

util.js
```js
export const foo = { bar: {} }
```

### Actual Behaviour

Rollup fails with:
```
[!] TypeError: Cannot read properties of null (reading 'getLiteralValueAtPath')
    at Identifier.getLiteralValueAtPath (<my-package>/node_modules/rollup/dist/shared/rollup.js:8423:47)
    at ChainExpression.getObjectValue (<my-package>/node_modules/rollup/dist/shared/rollup.js:10024:39)
    at ChainExpression.hasEffects (<my-package>/node_modules/rollup/dist/shared/rollup.js:10015:18)
    at VariableDeclarator.hasEffects (<my-package>/node_modules/rollup/dist/shared/rollup.js:12833:39)
    at VariableDeclaration.hasEffects (<my-package>/node_modules/rollup/dist/shared/rollup.js:6188:32)
    at Program.hasEffects (<my-package>/node_modules/rollup/dist/shared/rollup.js:11884:22)
    at Program.shouldBeIncluded (<my-package>/node_modules/rollup/dist/shared/rollup.js:6280:62)
    at Module.include (<my-package>/node_modules/rollup/dist/shared/rollup.js:13785:22)
    at Graph.includeStatements (<my-package>/node_modules/rollup/dist/shared/rollup.js:24490:36)
    at Graph.build (<my-package>/node_modules/rollup/dist/shared/rollup.js:24416:14)
```

[Removing the optional chaining](https://rollupjs.org/repl/?version=3.10.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHV0aWwlMjBmcm9tJTIwJy4lMkZ1dGlsJyUzQiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMHV0aWwuZm9vLmJhciUzQiU1Q25jb25zb2xlLmxvZyhiYXIpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydXRpbC5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMCU3QiUyMGJhciUzQSUyMCU3QiU3RCUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE) makes it succeed.

This [works with 3.9.1](https://rollupjs.org/repl/?version=3.9.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHV0aWwlMjBmcm9tJTIwJy4lMkZ1dGlsJyUzQiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMHV0aWwuZm9vJTNGLmJhciUzQiU1Q25jb25zb2xlLmxvZyhiYXIpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydXRpbC5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGZvbyUyMCUzRCUyMCU3QiUyMGJhciUzQSUyMCU3QiU3RCUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE), so I'm guessing it was caused by #4797. Root cause may be related to that of #4806.

**Comment by lukastaegert** [comment]
Fix at #4812

**Comment by ecraig12345** [comment]
Thanks for the quick fix!
